SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -172,6 +172,28 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 
 	TArray<UProperty*> RepNotifies;
 
+	// If we are applying initial data, they must have come from a ComponentData (as it currently stands).
```

Comment:
Shouldn't there be a simpler way to do this? Aren't we just saying if this is initial data, apply *ever* id (0->numids)?

Reply:
Spoke offline - the reason we are doing this weird loop is to gather all the fieldIds for a particular component (e.g. ActorData, or ActorOwnerOnly). We only want to apply the data of the component we got, but we want to apply all ids, even the missing ones. This loop is the way I do it - basically looking at the replication condition of the parent command to determine whether it would have gone into Data or OwnerOnly.